src/scripting.md

```diff
@@ -5,66 +5,179 @@ eleventyNavigation:
+
+A line that starts with  `// ` is ignored. Use this for comments.
+
+```markup
+// Example comment
+navigate not-a-comment.com
+```
+
+### Selectors
+
+Commands like [`click`](#click), [`setValue`](#setvalue), and others operate on DOM elements. To select a DOM element, use the following selectors:
+
+#### `<attribute>=<value>`
+
+This selector selects elements based on their HTML attributes. For the following input:
 
-For form fields it is usually best to use the name attribute when it is available since that is what will be uploaded to the server but any attribute is fair game. The class attribute is special and is referenced as className instead of class. In addition to the DOM element attribute matching, there are two special attributes that you can use to match on the contents. innerText and innerHtml both of which will match the contents of the DOM element instead of it's attributes.
-For example:
 ```html
-<div dojoattachpoint="containerNode" class="label">Delete</div>
+<input type="email" id="emailField" class="input input-email">
 ```
-Can be identified by innerText=Delete. The matching is case sensitive and matches the full string.
 
-In order to suppress intermediate steps you need to make sure that data logging is disabled for the steps up to the ones you want to record. For example:
+the following selectors will work:
+
+* ✅ `type=email`
+* ✅ `id=emailField`
+
+and the following selectors will not:
+
+* ❌ `class=input input-email` (spaces inside attribute values are not supported)
+* ❌ `class="input input-email"` (not even if quoted)
+
+Example:
 
 ```markup
-logData    0
+// Set the email to darth@vader.com
+setValue type=email darth@vader.com
+```
 
-// put any urls you want to navigate
-navigate    www.aol.com
-navigate    news.aol.com
+#### `innerText=<text>`
 
-logData    1
+This selector selects elements based on its inner text. For the following button:
 
-// this step will get recorded
-navigate    news.aol.com/world
+```html
+<button type="submit">Login</button>
 ```
 
-The script above will navigate to the main aol portal, then to the news page and then finally to the world-news specific page (logging results for just the world news page). This lets you test the impact of a given path to a site on it's performance (shared css and js caching for example).
+the following selectors will work:
+
+* ✅ `innerText=Login`
+
+and the following selectors will not:
 
-Another significant use case is if you want to test a site that requires authentication. Here is what an authentication script would look like:
+* ❌ `innerText=login` (the selector is case sensitive)
+* ❌ `innerText=log` (the selector must match the full string)
+
+Also, for a different button, this will not work either:
+
+* ❌ `innerText=log in` (spaces inside the text are not supported)
+
+Example:
 
 ```markup
-logData	0
+// Click the Login button
+click innerText=Login
+```
 
-// bring up the login screen
-navigate	http://webmail.aol.com
+#### `innerHtml=<html>`
```

Comment:
Is it `innerHtml` or `innerHTML`? The old doc used `innerHtml` but the DOM API is `innerHTML` so I’m not sure that wasn’t a typo.